### PR TITLE
feat(core/date-dropdown): new date-dropdown component

### PR DESCRIPTION
**Pull request by matthiashader**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Unit tests (`yarn test`) were run locally and passed
- [x] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [x] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bug fix
- [x] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

- date range selection via dropdown
- uses dayjs
- user can pass custom ranges

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

File: packages/core/src/components/date-dropdown/date-dropdown.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+@import 'common-variables';
+@import 'mixins/shadow-dom/component';
+
+:host {
+  @include ix-component;
+  display: block;
+
+  @media screen and (max-width: 500px) {
```
[the remark this answers is not in the document]
Comment (matthiashader) [reply packages/core/src/components/date-dropdown/date-dropdown.scss]: resolved

[the remark this answers is not in the document]
Comment (matthiashader) [reply packages/core/src/components/date-dropdown/date-dropdown.scss]: resolved

File: packages/core/src/components/date-dropdown/date-dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+   */
+  @Prop() range: boolean = true;
+
+  /**
+   * Picker date. If the picker is in range mode this property is the start date.
+   * If set to `null` no default start date will be pre-selected.
+   *
+   * Format is based on `format`
+   *
+   * @since 1.1.0
```
[the remark this answers is not in the document]
Comment (matthiashader) [reply packages/core/src/components/date-dropdown/date-dropdown.tsx]: resolved

File: packages/core/src/components/date-dropdown/date-dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+  from: dayjs.Dayjs;
+  to: dayjs.Dayjs;
+};
+
+export type DateRangeChangeEvent = {
+  from: string;
+  to: string;
+};
+
+@Component({
```
[the remark this answers is not in the document]
Comment (matthiashader) [reply packages/core/src/components/date-dropdown/date-dropdown.tsx:39]: resolved

File: packages/core/src/components/date-dropdown/date-dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+   * Used to set the initial select date range as well as the button name,
+   * if not set or no according date range label is found, nothing will be selected
+   * @default ''
+   */
+  @Prop() public initialSelectedDateRangeName: string = '';
+
+  /**
+   * Controls whether the user is allowed to pick custom date ranges in the component.
+   * When set to 'true', the user can select a custom date range using the date picker.
+   * When set to 'false', only predefined time date range are available for selection.
```
[the remark this answers is not in the document]
Comment (matthiashader) [reply packages/core/src/components/date-dropdown/date-dropdown.tsx]: resolved

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
Pressing 'Enter' when 'Custom..' is selected will close the dropdown. Datepicker should open instead.

File: packages/core/src/components/date-dropdown/date-dropdown.tsx [hunk 5]
```diff
@@ -188,10 +188,14 @@ export class DateDropdown {
...
       const { from, to } = getDateRange();
 
+      console.log(from);
+      console.log(to);
+
       this.associatedDateRangeValue = {
         from: from ? from.format(this.format) : undefined,
         to: to ? to.format(this.format) : undefined,
       };
+      console.log(this.associatedDateRangeValue);
```
Comment (nuke-ellington) [inline packages/core/src/components/date-dropdown/date-dropdown.tsx:198]:
```suggestion
```

File: packages/core/src/components/date-dropdown/date-dropdown.tsx [hunk 6]
```diff
@@ -188,10 +188,14 @@ export class DateDropdown {
     if (dateRangeName !== CUSTOM_RANGE_LABEL) {
       const { from, to } = getDateRange();
 
+      console.log(from);
```
Comment (nuke-ellington) [inline packages/core/src/components/date-dropdown/date-dropdown.tsx:191]:
```suggestion
```

File: packages/core/src/components/date-dropdown/date-dropdown.tsx [hunk 7]
```diff
@@ -188,10 +188,14 @@ export class DateDropdown {
...
+      console.log(to);
```
Comment (nuke-ellington) [inline packages/core/src/components/date-dropdown/date-dropdown.tsx:192]:
```suggestion
```

File: packages/core/src/components/date-dropdown/date-dropdown.tsx [hunk 8]
```diff
@@ -188,10 +188,14 @@ export class DateDropdown {
...
+
```
Comment (nuke-ellington) [inline packages/core/src/components/date-dropdown/date-dropdown.tsx:193]: Sorry cannot do multiline suggestion for some reason today :/

```suggestion
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

**Review by nuke-ellington — APPROVED** [review]

**Review by nuke-ellington — APPROVED** [review]

**Comment by robert-wettstaedt** [comment]
Hello, are there any plans on when this new feature is going to be released in a new version? It's been merged for almost 3 months now.

**Comment by matthiashader** [comment]
@robert-wettstaedt sorry for keeping you waiting. This component was released with v2.1.0

**Pull request merged**